src/tracking.c

```diff
@@ -153,15 +155,17 @@ int checkPrefixCollisionsOrReply(client *c, robj **prefixes, size_t numprefix) {
 /* Set the client 'c' to track the prefix 'prefix'. If the client 'c' is
  * already registered for the specified prefix, no operation is performed. */
 void enableBcastTrackingForPrefix(client *c, char *prefix, size_t plen) {
-    bcastState *bs = raxFind(PrefixTable,(unsigned char*)prefix,plen);
+    void *result;
     /* If this is the first client subscribing to such prefix, create
      * the prefix in the table. */
-    if (bs == raxNotFound) {
-        bs = zmalloc(sizeof(*bs));
+    if (!raxFind(PrefixTable,(unsigned char*)prefix,plen,&result)) {
+        bcastState *bs = zmalloc(sizeof(*bs));
         bs->keys = raxNew();
         bs->clients = raxNew();
         raxInsert(PrefixTable,(unsigned char*)prefix,plen,bs,NULL);
+        result = bs;
     }
+    bcastState *bs = result;
```

Comment:
Similar here, two variables with the same name and assignement back and forth. Maybe we can rewrite it with `else` like my other suggestions, if you like that suggestion.

Reply:
Thanks Vitkor. I agree with your comments that a variable name at different nesting levels is confusing when reading back. Have fixed with in an additional commit now.